src/v/cluster/persisted_stm.h

```diff
@@ -79,6 +79,18 @@ class file_backed_stm_snapshot {
     size_t _snapshot_size{0};
 };
 
+template<typename T>
+concept supported_stm_backing_store = requires(T s, stm_snapshot&& snapshot) {
```

Comment:
nit: maybe something like `persisted_stm_snapshot_backend` would be more readable